src/decimal-field.tsx

```diff
@@ -37,25 +56,6 @@ export const DecimalInput: React.FC<DecimalInputProps> = ({
     const {i18n} = useTheme();
     const localDecimalChar = getLocalDecimalChar(i18n.locale);
 
-    const format = (value: any) => {
-        const parts = String(value ?? '')
-            .replace(/[^.,\d]/g, '') // remove non digits or decimal separator chars
-            .split(localDecimalChar);
-
-        if (parts.length === 0) {
-            // empty
-            return '';
-        }
-
-        if (parts.length === 1) {
-            // no fractional part, return "as is"
-            return parts[0];
-        }
-
-        // value includes one or more decimal separators, keep the first one
-        return parts.shift() + localDecimalChar + parts.join('');
-    };
-
```

Comment:
The bug arose because I deleted the line which made the replacement by the localDelimiterChar. That replacement prevented rifm from working properly, that is why I moved it to rifm's replace input. It seems that rifm doesn't accept replacements in format functions. 

I've moved format function outside the component since localDecimalChar is no longer necessary in it. This way, in future revisions, developers won't be tempted to make that replacement in the format function

Reply:
I'd also include a comment explaining this

Reply:
will this do?? https://github.com/Telefonica/mistica-web/pull/306/files#diff-9aec20dcbd46eed10b4693dec90d2e2a9970b54e570ff4b38f2b20495eaa622cR21-R22